### PR TITLE
Handle track saving in non-album contexts (eg. playlist/Made for you).

### PR DESCRIPTION
**Pull request by jackson15j**
Saving of tracks used to only work in album view (handled by:
`src.handlers.album_tracks.handle_save_album_event`).

Add some rough code to get track saving working in other cases. Manually tested
`Playlist` and `Made for you` views.

Fixes Rigellute/spotify-tui#495

### CODE REVIEW

**Comment by jackson15j** [comment]
Apologies if the code is dirty. First time looking at Rust. Let me know what additional work you'd like me to do to progress this further.

File: src/handlers/track_table.rs [hunk 1]
```diff
@@ -252,6 +253,16 @@ fn play_random_song(app: &mut App) {
   };
 }
 
+fn handle_save_track_event(app: &mut App) {
+  let (selected_index, tracks) = (&app.track_table.selected_index, &app.track_table.tracks);
+  if let Some(track) = tracks.get(*selected_index) {
+    if let Some(id) = &track.id {
+      let _id = id.to_string();
+      app.dispatch(IoEvent::ToggleSaveTrack(_id));
```
Comment (Rigellute) [inline src/handlers/track_table.rs:261]: The underscore prefix (`_id`) is usually used to convey that the variable is not used.

Instead you could do this?

```suggestion
      app.dispatch(IoEvent::ToggleSaveTrack(id.to_string()));
```

Comment (jackson15j) [reply src/handlers/track_table.rs:261]: Good to know the underscore means unused (Bad habit from other languages, where the underscore prefix is for a private/local variable).

Your suggestion is what I had in the original commit, which broke the build checks.

Comment (Rigellute) [reply src/handlers/track_table.rs:261]: Ah okay cool, that's all good then!

So if you could just remove the underscore then we are good to go!

Comment (jackson15j) [reply src/handlers/track_table.rs]: Fixed up and also merged in latest master changes. Works locally. So hoping all Github checks still pass.

**Comment by Rigellute** [comment]
Thanks for your contribution @jackson15j. 

As a side note, it is typically preferable to use `git rebase` to sync a branch with master, that way we avoid the merge commit we see above. I need to create a contributors guide detailing this, so no worries for now!

@all-contributors please add @jackson15j for code.

**Pull request merged**